Realm/RLMQueryUtil.mm

```diff
@@ -762,6 +822,27 @@ void validate_and_extract_between_range(id value, RLMProperty *prop, id *from, i
     m_query.end_group();
 }
 
+void QueryBuilder::add_binary_substring_constraint(size_t index,
+                                                   StringData value,
+                                                   Query&(Query::*constraint_maker)(size_t, StringData, bool)) {
+    // Foundation always returns false for substring operations with a RHS of null or "".
+    if (value.size()) {
+        (m_query.*constraint_maker)(index, std::move(value), true);
```

Comment:
Unfortunately, because of the differing arities of the string and binary versions of this `Query` API, I don't think there's a way to collapse these two overloads.